src/scripts/testbot/guardrails.py

```
@@ -0,0 +1,68 @@
+    for file_path in sorted(all_files):
+        if is_test_file(file_path):
+            test_files.append(file_path)
+        else:
+            non_test_files.append(file_path)
+
+    if non_test_files:
+        logger.warning(
+            "Discarding %d non-test file(s) modified by Claude: %s",
+            len(non_test_files), non_test_files,
+        )
+        subprocess.run(["git", "checkout", "--"] + non_test_files, check=False)
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)